sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcFileFormat.scala

```diff
@@ -179,12 +179,17 @@ class OrcFileFormat
 
       val fs = filePath.getFileSystem(conf)
       val readerOptions = OrcFile.readerOptions(conf).filesystem(fs)
-      val requestedColIdsOrEmptyFile =
+      val (requestedColIdsOrEmptyFile, sendActualSchema) =
         Utils.tryWithResource(OrcFile.createReader(filePath, readerOptions)) { reader =>
           OrcUtils.requestedColumnIds(
             isCaseSensitive, dataSchema, requiredSchema, reader, conf)
         }
 
+      if (sendActualSchema) {
+        resultSchemaString = OrcUtils.orcTypeDescriptionString(actualSchema)
```

Comment:
do you mean we can't do column pruning in this case?

Reply:
Yes for this case we cannot do it ,so as the result of which we have the give the actual schema